generator/src/lib.rs

```diff
@@ -1797,9 +1797,13 @@ pub fn derive_setters(
 
             if name == "ppGeometries" {
                 return Some(quote!{
-                    pub fn geometries_ptrs(mut self, geometries: &'a [*const AccelerationStructureGeometryKHR]) -> Self {
+                    pub fn geometries_ptrs(mut self, geometries: &'a [&'a AccelerationStructureGeometryKHR]) -> Self {
                         self.inner.geometry_count = geometries.len() as _;
-                        self.inner.pp_geometries = geometries.as_ptr();
+                        assert_eq!(
+                            std::mem::size_of::<*const AccelerationStructureGeometryKHR>(),
+                            std::mem::size_of::<&'a AccelerationStructureGeometryKHR>()
+                        );
```

Comment:
If only we had compile-time asserts without an external crate...

Reply:
I'm not sure this assert is justified at all, really. `AccelerationStructureGeometryKHR` is never going to be a DST, and I don't think rust references will ever not be ABI-compatible with dereffable pointers.

Reply:
Apologies for the paranoia. This won't be my first time debugging raw bindings where a DST (slice or dynamic-dispatch object) slips in and ruins the data going across. If only `as *const` wouldn't compile when converting a 16-byte fat pointer to a raw pointer that'd be so much better. Removed it again since this pattern is pretty well used across Ash anyway.